### DISCUSSION TITLE
bme280 lost after enabling mqtt and creating rule from console

### DISCUSSION THREAD

**Discussion by mesiu84** [arendst/Tasmota#20075] [Q&A] [answered]
### PROBLEM DESCRIPTION
BME280 is lost from Tasmota after doing following:
1. clear setup
2. configure bme280 in UI + restart <- all works OK
3. configure MQTT + restart <- all works OK
4. add internal rule (rule1 on bme280#humidity>70 do power1 1 endon on bme280#humidity<=70 do power1 0 endon) <- all works OK
5. enable rule1 (rule1 1) <- all works OK
6. restart tasmota

After doing above steps bme280 is lost and I don't see any readings from it. Doing restart doesn't fix the issue, but if I do:
7. rule1 0
8. restart tasmota

Then bme280 is back again. Enabling rule1 (2 or 3 doesn't matter) and doing a restart I can't see bme280 again, if I don't do a restart all works until next restart or there is no power and tasmota boots up again

### REQUESTED INFORMATION
  Configuration output here:
Sonoff Basic R2
Program Version	13.2.0(sensors) <- from official URL
Core/SDK Version	2_7_4_9/2.2.2-dev(38a443e)
Uptime	2T18:12:35
Flash Write Count	833 at 0xF5000
Boot Count	14

  Rules output here:
```17:56:33.937 MQT: stat/tasmota/RESULT = {"Rule1":{"State":"ON","Once":"OFF","StopOnError":"OFF","Length":80,"Free":431,"Rules":"on bme280#humidity>70 do power1 1 endon on bme280#humidity<=70 do power1 0 endon"}}```

  STATUS 0 output here:
```17:53:48.843 CMD: status 0
17:53:48.852 MQT: stat/tasmota/STATUS = {"Status":{"Module":1,"DeviceName":"Tasmota","FriendlyName":["Tasmota"],"Topic":"tasmota","ButtonTopic":"0","Power":0,"PowerOnState":3,"LedState":1,"LedMask":"FFFF","SaveData":1,"SaveState":1,"SwitchTopic":"","SwitchMode":[1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],"ButtonRetain":0,"SwitchRetain":0,"SensorRetain":0,"PowerRetain":0,"InfoRetain":0,"StateRetain":0,"StatusRetain":0}}
17:53:48.868 MQT: stat/tasmota/STATUS1 = {"StatusPRM":{"Baudrate":115200,"SerialConfig":"8N1","GroupTopic":"tasmotas","OtaUrl":"http://ota.tasmota.com/tasmota/release/tasmota.bin.gz","RestartReason":"Software/System restart","Uptime":"2T18:15:08","StartupUTC":"2023-11-17T22:38:40","Sleep":50,"CfgHolder":4617,"BootCount":14,"BCResetTime":"2023-11-17T17:51:45","SaveCount":833,"SaveAddress":"F5000"}}
17:53:48.880 MQT: stat/tasmota/STATUS2 = {"StatusFWR":{"Version":"13.2.0(sensors)","BuildDateTime":"2023-10-19T09:02:00","Boot":31,"Core":"2_7_4_9","SDK":"2.2.2-dev(38a443e)","CpuFrequency":80,"Hardware":"ESP8285N08","CR":"380/699"}}
17:53:48.892 MQT: stat/tasmota/STATUS3 = {"StatusLOG":{"SerialLog":2,"WebLog":2,"MqttLog":0,"SysLog":0,"LogHost":"","LogPort":514,"SSId":["doma",""],"TelePeriod":300,"Resolution":"558180C0","SetOption":["00008009","2805C80001000600003C5A0A002800000000","00000080","00006000","00004000","00000000"]}}
17:53:48.920 MQT: stat/tasmota/STATUS4 = {"StatusMEM":{"ProgramSize":701,"Free":300,"Heap":14,"ProgramFlashSize":1024,"FlashSize":1024,"FlashChipId":"14325E","FlashFrequency":40,"FlashMode":"DOUT","Features":["00000809","8F9A8787","0415A005","B7F7BFCF","05DA9BC4","64367CC7","00084052","20000000","54000020","0008C081"],"Drivers":"1,2,3,4,5,6,7,8,9,10,12,14,16,17,20,21,24,29,34,62,65,66,68","Sensors":"1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,17,18,19,20,21,22,26,31,34,37,39,40,42,43,45,51,52,55,56,58,59,62,64,66,67,74,98,103,105","I2CDriver":"7,8,9,10,11,12,13,14,15,17,18,20,24,29,31,36,41,42,44,46,48,69,76"}}
17:53:48.939 MQT: stat/tasmota/STATUS5 = {"StatusNET":{"Hostname":"tasmota-2493","IPAddress":"192.168.1.188","Gateway":"192.168.1.1","Subnetmask":"255.255.255.0","DNSServer1":"192.168.1.1","DNSServer2":"0.0.0.0","Mac":"48:55:19:77:C9:BD","Webserver":2,"HTTP_API":1,"WifiConfig":4,"WifiPower":17.0}}
17:53:48.954 MQT: stat/tasmota/STATUS6 = {"StatusMQT":{"MqttHost":"192.168.1.104","MqttPort":1883,"MqttClientMask":"DVES_77C9BD","MqttClient":"DVES_77C9BD","MqttUser":"mqtt-user","MqttCount":4,"MAX_PACKET_SIZE":1200,"KEEPALIVE":30,"SOCKET_TIMEOUT":4}}
17:53:48.971 MQT: stat/tasmota/STATUS7 = {"StatusTIM":{"UTC":"2023-11-20T16:53:48","Local":"2023-11-20T17:53:48","StartDST":"2023-03-26T02:00:00","EndDST":"2023-10-29T03:00:00","Timezone":"+01:00","Sunrise":"08:06","Sunset":"17:04"}}
17:53:48.989 MQT: stat/tasmota/STATUS10 = {"StatusSNS":{"Time":"2023-11-20T17:53:48","BME280":{"Temperature":21.6,"Humidity":64.3,"DewPoint":14.5,"Pressure":982.0},"PressureUnit":"hPa","TempUnit":"C"}}
17:53:49.002 MQT: stat/tasmota/STATUS11 = {"StatusSTS":{"Time":"2023-11-20T17:53:48","Uptime":"2T18:15:08","UptimeSec":238508,"Heap":16,"SleepMode":"Dynamic","Sleep":50,"LoadAvg":19,"MqttCount":4,"POWER":"OFF","Wifi":{"AP":1,"SSId":"WiFiNet","BSSId":"9C:3D:CF:CA:7B:C0","Channel":13,"Mode":"11n","RSSI":68,"Signal":-66,"LinkCount":1,"Downtime":"0T00:00:03"}}}```


### EXPECTED BEHAVIOUR
BME280 should not disappear

**Comment by sfromis** [comment]
You did not post much of the requested info, specifically detailed logs from when it happens.

Anyway, using Sonoff Basic R2 is just about the worst hardware you can find, not a good platform for adding stuff to, and you are not the first to have trouble running the sensors build on that thing. You may have a boot loop causing Tasmota to "scale back" features until able to boot without crashing quickly again. It tends to work better using decent hardware instead.

Maybe you'd have luck picking up a very old Tasmota version with less features. Or maybe do a custom build to only include features you want.

**Comment by mesiu84** [comment]
That's actually my plan to test older version (8.1.0 is my target) and if that doesn't work to try my custom build. But what would be a better hardware to work with? Sonoff was the easiest to get for me :-) but now I can try something else too
plus - thare are no logs since I'm doing a reboot, it just doesn't find the sensor after reboot, i2cscan also doesn't detect anything
I was hoping someone had similar issue, but as you said, probably lowering the version or custom build would solve the issue, I would try to test it

**Comment by sfromis** [comment]
Since you got the worst device, the Sonoff Basic R2, "better" means "something else", which does not exclude Sonoff devices.

And you could have gotten better info if you did not erase it by doing a reboot.

More generally, what I do for sensor devices is not to add stuff to unsuitable devices, but use dev boards with freely accessible pins.

**Comment by mesiu84** [comment]
I don't need anything fancy, it's just a small thing that I'm doing, that's why I don't need any extra devices. The question is more for a future, what is better in your opinion? I'm mostly using zigbee protocol and devices that work with zigbee, that sonoff was something that I just didn't found anywhere with zigbee.

**Comment by Wintespe** [comment, marked as answer]
There are 2 different Sonoff Basic R2's. Both are marked "R2" but are from different years. 


![R2 old](https://github.com/arendst/Tasmota/assets/121858068/9057eab1-28c8-48ee-a1b2-c70179a6d246)


![R2 new](https://github.com/arendst/Tasmota/assets/121858068/b3e5caae-b7a9-4e58-8267-e59462af7c6e)




The older one is labeld 2017 and the newer one with 1920. Which one has the power problem?

**Comment by mesiu84** [comment]
Would check that tomorrow, for now I've build my custom version of tasmota where I removed all that I don't have/use/need and now I don't have the problem anymore, but I'll check the version and update it here.

**Comment by mesiu84** [comment]
Here is the version I have, v1.3 2022.03.09
![sonoff_basic_r2](https://github.com/arendst/Tasmota/assets/8832765/d592848d-02bc-4b17-b335-16e4d2d2a2a9)

**Comment by Wintespe** [comment]
The Sonoff "R2" problem has been the subject of discussion for several years. So far it is unclear to me which Sonfoffs are really affected by the "worst hardware you can find". I hope that some light will now be shed on the matter

**Reply by sfromis** [reply]
Seeing many users getting into trouble suggests that "Sonoff Basic R2" is damn hard to recommend. Specific board revisions have not been mentioned much either way, and it would be hard to know which revision some seller may have in stock. Anecdotes about someone having experienced no trouble with some specific one does not count for much, as it seems to be a case of borderline worsened with aging. Also the original "Sonoff Basic" is by now very often seen as a troublemaker, and while the R2 was pointed out as worse a long time ago, aging of caps may be a significant factor.

Still, I've seen suggestions that there may have been improvements; not sure how much of this is down to "younger".

**Comment by mesiu84** [comment]
As long as the compiled version works for me I'm fine, I'm not going to update it soon :-) I'm closing this discussion then
